tests/specs/functional/merkle-spec.k

```diff
@@ -3,20 +3,21 @@ requires "evm.md"
 module VERIFICATION
     imports EVM
 
-    syntax StepSort ::= MerkleTree | String
+    syntax StepSort ::= MerkleTree | String | Int
     syntax    KItem ::= runMerkle ( StepSort )
                       | doneMerkle( StepSort )
  // ------------------------------------------
-    rule runMerkle( T ) => doneMerkle( T )
+    rule <k> runMerkle( T ) => doneMerkle( T ) ... </k>
 
-    syntax MerkleTree ::= "#initTree" [function]
- // --------------------------------------------
+    syntax MerkleTree ::= "#initTree"
+ // ---------------------------------
     rule #initTree => MerkleUpdateMap( .MerkleTree,
-                                       #parseByteStack( "do" )    |-> "verb"
-                                       #parseByteStack( "dog" )   |-> "puppy"
-                                       #parseByteStack( "doge" )  |-> "coin"
-                                       #parseByteStack( "horse" ) |-> "stallion"
+                                       .Map [ #parseByteStack( "de" )    <- "verb"     ]
+                                            [ #parseByteStack( "def" )   <- "puppy"    ]
+                                            [ #parseByteStack( "defe" )  <- "coin"     ]
+                                            [ #parseByteStack( "becfe" ) <- "stallion" ]
                                      )
+      [macro]
```

Comment:
This just means that `#initTree` will be replaced with the `MerkleUpdateMap( ... )` after `kompile`, instead of making the backend evaluate `#initTree`, right?

Reply:
Yeah, basically. It also uses the map update operator instead of map concat, so all the appropriate definedness checks are triggered. It also uses actual hex strings instead of random stcrings, so that `#parseByteStack(_)` is defined as well.